File: MotorcycleRideLog/src/MotorcycleRideLog.WebApp/projects/motorcycle-ride-log/src/app/models/motorcycle.ts
import { MotorcycleType } from './motorcycle-type';

export interface Motorcycle {
  motorcycleId: string;
  userId: string;
  make: string;
  model: string;
  year?: number;
  type: MotorcycleType;
  vin?: string;
  licensePlate?: string;
  currentMileage?: number;
  color?: string;
  notes?: string;
  createdAt: string;
}

export interface CreateMotorcycle {
  userId: string;
  make: string;
  model: string;
  year?: number;
  type: MotorcycleType;
  vin?: string;
  licensePlate?: string;
  currentMileage?: number;
  color?: string;
  notes?: string;
}

export interface UpdateMotorcycle {
  motorcycleId: string;
  make: string;
  model: string;
  year?: number;
  type: MotorcycleType;
  vin?: string;
  licensePlate?: string;
  currentMileage?: number;
  color?: string;
  notes?: string;
}

File: MotorcycleRideLog/src/MotorcycleRideLog.Core/Model/MotorcycleAggregate/Motorcycle.cs
// Licensed to the .NET Foundation under one or more agreements.
// The .NET Foundation licenses this file to you under the MIT license.

namespace MotorcycleRideLog.Core;

public class Motorcycle
{
    public Guid MotorcycleId { get; set; }
    public Guid UserId { get; set; }
    public string Make { get; set; } = string.Empty;
    public string Model { get; set; } = string.Empty;
    public int? Year { get; set; }
    public MotorcycleType Type { get; set; }
    public string? VIN { get; set; }
    public string? LicensePlate { get; set; }
    public int? CurrentMileage { get; set; }
    public string? Color { get; set; }
    public string? Notes { get; set; }
    public DateTime CreatedAt { get; set; }
}
